docs/howto/openstack/keystone/app-creds.md

```diff
@@ -0,0 +1,310 @@
+---
+description: A walk-through of creating and managing application credentials
+---
+# Application credentials
+
+You can create application credentials and thus enable your applications to use them to authenticate against OpenStack Keystone.
+Thanks to this type of credentials, you do not have to resort to your user credentials.
+Applications can still authenticate using the application credentials ID along with a secret string, which has **nothing** to do with your user password.
+That way, this sensitive piece of information does not have to be embedded in any particular application configuration.
+
+One major security feature of application credentials is their capacity for expiration.
+More often than not, application credentials are made to automatically expire after a certain date and time.
+Passwords, on the other hand, persist essentially forever --- unless they are invalidated manually.
+
+## Prerequisites
+
+In {{brand}}, you manage application credentials with the OpenStack CLI, so make sure to [enable it](../../getting-started/enable-openstack-cli.md) for the region you are interested in.
+
+## Creating application credentials
+
+To create new application credentials with an expiration date, decide on a name and then use the `openstack` client like so:
+
+```console
+$ openstack application credential create --expiration '2024-04-30T12:00:00' angry_lamarr
+
++--------------+-----------------------------------------------------------------------------------+
+| Field        | Value                                                                             |
++--------------+-----------------------------------------------------------------------------------+
+| description  | None                                                                              |
+| expires_at   | 2024-04-30T12:00:00.000000                                                        |
+| id           | 98a2804b2d294b6784ae30ec67697382                                                  |
+| name         | angry_lamarr                                                                      |
+| project_id   | dfc700467396428bacba4376e72cc3e9                                                  |
+| roles        | creator reader swiftoperator load-balancer_member member                          |
+| secret       | BGXRFpTtEeh6Wi2MsohpFajz5V6ZVGMELjzH8gn53M1SD0voaOEYKlZsmtgO346FqONCHb0bEyt1wnUcv |
+|              | PYYzA                                                                             |
+| system       | None                                                                              |
+| unrestricted | False                                                                             |
+| user_id      | e50719d594f84f16ad13f88da540f762                                                  |
++--------------+-----------------------------------------------------------------------------------+
+```
+
```

Comment:
I think we should mention here that the `--expiration` argument format is an [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) date string (which implies that it can be abbreviated to just a date, and that one can add a timezone specifier if necessary).

Reply:
Yes, added a note about that!